src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/EnclaveSessionCache.cs

```diff
@@ -67,9 +67,9 @@ internal SqlEnclaveSession CreateSession(string attestationUrl, string serverNam
         }
 
         // Generates the cache key for the enclave session cache
-        private string GenerateCacheKey(string serverName, string attestationUrl)
+        private string GenerateCacheKey(string serverName, string attestationUrl, string database)
```

Comment:
Maybe use a parameters class instead of passing three strings?

Reply:
The strings we used are passed from multiple layers of function calls before. From my point of view, it's straightforward to keep using them here instead of wrapping them in an extra class before generating the cache key. Is there any benefit of using a class for this specific function?

Reply:
I think a good reason could be to avoid/make it less likely  passing parameters in the wrong order (since all parameters are strings)

Reply:
Thank you for your explanation. I understand the concerns. If we are going to wrap these strings, I think we need to do it before we pass them to the enclave delegate. It requires the corresponding changes in all enclave providers too.

Reply:
It was just a suggestion.

Reply:
We value your suggestion. The current APIs for enclave providers are inherited from System.Data.SqlClient.SqlColumnEncryptionEnclaveProvider interface. To avoid passing multiple strings for server name, attestation URL, and database name, the APIs need to be updated. It is doable for Microsoft.Data.SqlClient since we have made these APIs internal.

This PR is a hotfix for the GA-2.0.0 which introduces minor impacts on the released driver. I will bring up your suggestions within the team and will probably create a separate PR for the API changes if we are good to go with it.

Reply:
We will consider this design change in PR for 'master' branch.